<!--
 Copyright 2022 Shadow Robot Company Ltd.
 This program is free software: you can redistribute it and/or modify it
 under the terms of the GNU General Public License as published by the Free
 Software Foundation version 2 of the License.
 This program is distributed in the hope that it will be useful, but WITHOUT
 ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
 more details.
 You should have received a copy of the GNU General Public License along
 with this program. If not, see <http://www.gnu.org/licenses/>.
-->

<launch>
  <arg name="serial_port" default="/dev/ttyUSB0"/>
  <arg name="calibration" default="$(find sr_cyberglove_config)/calibrations/cyberglove.cal"/>
  <arg name="version" default="2"/>
  <arg name="protocol" default="8bit"/>
  <!-- Activate internal cybeglove data filtering -->
  <arg name="filter" default="true"/>

  <param name="robot_description" textfile="$(find
  cyberglove)/model/cyberglove.xml"/>

  <node pkg="cyberglove" name="cyberglove" type="cyberglove_node" >
    <param name="cyberglove_prefix" type="string" value="/cyberglove" />
    <!-- We're doing some oversampling. You can set the frequency at which
	 the data are sampled as well as the frequency at which they're published
	 possible values:
	   - 100 -> 100 Hz
	   - 45  -> 45 Hz
	   - 10  -> 10 Hz
	   - 1  -> 1 Hz
    -->
    <param name="sampling_frequency" type="double" value="100.0" />
    <param name="publish_frequency" type="double" value="100.0" />
    <param name="path_to_glove" type="string" value="$(arg serial_port)" />
    <param name="path_to_calibration" type="string" value="$(arg calibration)" />
    <param name="cyberglove_version" type="string" value="$(arg version)" />
    <param name="streaming_protocol" type="string" value="$(arg protocol)" />
    <param name="filter" type="bool" value="$(arg filter)" />
  </node>

  <!-- Robot state publisher -->
  <!-- node pkg="robot_state_publisher" type="state_publisher"
	name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="10.0" />
    <param name="tf_prefix" type="string" value="" />
  </node-->
</launch>
